File: utils/embed_manager.py
import json
import discord
from typing import Dict, Optional
import os

class EmbedManager:
    def __init__(self):
        self.cache: Dict[str, discord.Embed] = {}
        self.templates = self._load_templates()
        
    def _load_templates(self) -> dict:
        path = os.path.join(os.path.dirname(__file__), '..', 'data', 'default_embeds.json')
        try:
            with open(path, 'r') as f:
                return json.load(f)
        except FileNotFoundError:
            return {}
            
    def get_template(self, template_name: str) -> Optional[dict]:
        return self.templates.get(template_name)
        
    def create_embed(self, template_name: str = None, **kwargs) -> discord.Embed:
        if template_name and template_name in self.cache:
            # Return a copy of cached embed
            return discord.Embed.from_dict(self.cache[template_name].to_dict())
            
        # Get template or empty dict if template doesn't exist
        template = self.templates.get(template_name, {})
        
        # Merge template with kwargs
        embed_data = {**template, **kwargs}
        
        embed = discord.Embed(
            title=embed_data.get('title', ''),
            description=embed_data.get('description', ''),
            color=embed_data.get('color', 0)
        )
        
        if embed_data.get('footer'):
            embed.set_footer(text=embed_data['footer'])
            
        if embed_data.get('timestamp', False):
            embed.timestamp = discord.utils.utcnow()
            
        # Cache the template if it's a template embed
        if template_name:
            self.cache[template_name] = embed
            
        return embed

File: utils/embedManager.js
const { EmbedBuilder } = require('discord.js');
const fs = require('fs').promises;
const path = require('path');

class EmbedManager {
    constructor() {
        this.cache = new Map();
        this.templates = {};
        this._loadTemplates();
    }

    async _loadTemplates() {
        try {
            const filePath = path.join(__dirname, '..', 'data', 'default_embeds.json');
            const data = await fs.readFile(filePath, 'utf8');
            this.templates = JSON.parse(data);
        } catch (error) {
            if (error.code !== 'ENOENT') {
                console.error('Error loading templates:', error);
            }
            this.templates = {};
        }
    }

    getTemplate(templateName) {
        return this.templates[templateName] || null;
    }

    createEmbed(templateName = null, options = {}) {
        if (templateName && this.cache.has(templateName)) {
            return EmbedBuilder.from(this.cache.get(templateName));
        }

        const template = templateName ? this.getTemplate(templateName) || {} : {};
        const embedData = { ...template, ...options };

        const embed = new EmbedBuilder()
            .setTitle(embedData.title || '')
            .setDescription(embedData.description || '')
            .setColor(embedData.color || 0);

        if (embedData.footer) {
            embed.setFooter({ text: embedData.footer });
        }

        if (embedData.timestamp) {
            embed.setTimestamp();
        }

        if (templateName) {
            this.cache.set(templateName, embed);
        }

        return embed;
    }
}

module.exports = EmbedManager;
